src/librustc/diagnostics.rs

```diff
@@ -2043,6 +2043,23 @@ a (non-transparent) struct containing a single float, while `Grams` is a
 transparent wrapper around a float. This can make a difference for the ABI.
 "##,
 
+E0698: r##"
+When using generators (or async) all type variables must be specified so a
```

Comment:
We probably want to follow [the standard conventions here](https://rust-lang.github.io/rfcs/1567-long-error-codes-explanation-normalization.html). I feel like if I got this error message, it wouldn't help me figure out what to change -- it'd be good to highlight a fixed version of the example so people can see the difference.